core/drivers/stm32_gpio.c

```diff
@@ -933,21 +1209,146 @@ static TEE_Result stm32_pinctrl_dt_get(struct dt_pargs *pargs,
 }
 #endif /*CFG_DRIVERS_PINCTRL*/
 
+static void stm32_gpio_get_conf_sec(struct stm32_gpio_bank *bank)
+{
+	if (bank->sec_support) {
+		clk_enable(bank->clock);
+		bank->seccfgr = io_read32(bank->base + GPIO_SECR_OFFSET);
+		clk_disable(bank->clock);
+	}
+}
+
+static void stm32_gpio_set_conf_sec(struct stm32_gpio_bank *bank)
+{
+	if (bank->sec_support) {
+		clk_enable(bank->clock);
+		io_write32(bank->base + GPIO_SECR_OFFSET, bank->seccfgr);
+		clk_disable(bank->clock);
+	}
+}
+
+static TEE_Result stm32_gpio_sec_config_resume(void)
+{
+	struct stm32_gpio_bank *bank = NULL;
+
+	STAILQ_FOREACH(bank, &bank_list, link) {
+		if (bank->rif_cfg) {
+			if (!bank->is_tdcid)
+				continue;
+
+			bank->rif_cfg->access_mask[0] = GENMASK_32(bank->ngpios,
+								   0);
+
+			apply_rif_config(bank);
```

Comment:
I think we should test the return value here, to address case when for example we fail to get a semaphore we need.